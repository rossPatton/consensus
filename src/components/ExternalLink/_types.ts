export type tProps = {
  children: React.ReactNode,
  className?: string,
  noFollow: boolean,
  to: string,
};
