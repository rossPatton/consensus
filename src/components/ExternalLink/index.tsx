import React, { memo } from 'react';

import { tProps } from './_types';

const ExternalLink = memo((props: tProps) => (
  <a
    href={props.to}
    className={props.className || ''}
    rel={`noopener noreferrer ${props.noFollow ? 'nofollow' : ''}`}>
    {props.children}
  </a>
));

export default ExternalLink;
