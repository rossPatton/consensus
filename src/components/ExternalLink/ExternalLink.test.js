import React from 'react';
import render from 'react-test-renderer';

import ExternalLink from '.';

it('renders without crashing', () => {
  render.create(<ExternalLink />);
});

it('renders as expected (simple)', () => {
  const component = render.create((
    <ExternalLink>
      test
    </ExternalLink>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders as expected (className/noFollow)', () => {
  const component = render.create((
    <ExternalLink
      noFollow
      className="test">
      test
    </ExternalLink>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
