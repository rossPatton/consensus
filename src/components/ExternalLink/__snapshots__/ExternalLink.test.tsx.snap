// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected (className/noFollow) 1`] = `
<a
  className="test"
  href="https://www.test.com"
  rel="noopener noreferrer nofollow"
>
  test
</a>
`;

exports[`renders without crashing 1`] = `
<a
  className=""
  href="https://www.test.com"
  rel="noopener noreferrer "
>
  test
</a>
`;
